### PR TITLE
[SR-1822] Add new fixits for oldhat 'inout'

### PR DESCRIPTION
**Pull request by CodaFi**
#### What's in this pull request?

Mirrors the fixits provided in ParsePattern for tuples.  We now
diagnose multiple extraneous uses of inout and uses of inout attached
to the parameter name and not the type and offer to move them.
#### Resolved bug number: ([SR-1822](https://bugs.swift.org/browse/SR-1822))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please test

File: test/Parse/invalid.swift [hunk 1]
```diff
@@ -69,8 +70,8 @@ SR698(1, b: 2,) // expected-error {{unexpected ',' separator}}
 
 // SR-979 - Two inout crash compiler
 func SR979a(a : inout inout Int) {}  // expected-error {{parameter may not have multiple 'inout', 'var', or 'let' specifiers}} {{17-23=}}
-func SR979b(inout inout b: Int) {}  // expected-error {{parameter may not have multiple 'inout', 'var', or 'let' specifiers}} {{19-25=}}
-// expected-error @-1 {{'inout' before a parameter name is not allowed, place it before the parameter type instead}} {{13-18=}} {{28-28=inout }}
+func SR979b(inout inout b: Int) {} // expected-error {{inout' before a parameter name is not allowed, place it before the parameter type instead}}
```
Comment (jrose-apple) [inline test/Parse/invalid.swift]: What happened to the fix-its?

File: lib/Parse/ParseType.cpp [hunk 2]
```diff
@@ -491,13 +508,18 @@ ParserResult<TupleTypeRepr> Parser::parseTypeTupleBody() {
       if (type.isNull())
         return makeParserError();
 
+      if (!hasValidInOut && hasDeprecatedInOut) {
```
Comment (jrose-apple) [inline lib/Parse/ParseType.cpp]: What about when both are present?

{{(inout a: inout Int) -> Bool}}

File: test/Parse/invalid.swift (hunk 1 above)
Comment (CodaFi) [inline test/Parse/invalid.swift]: Whoops.  They're back.

File: lib/Parse/ParseType.cpp (hunk 2 above)
Comment (CodaFi) [inline lib/Parse/ParseType.cpp]: The fixit offers to remove the invalid one.

**Comment by CodaFi** [comment]
@swift-ci please test and merge.

**Pull request merged**